apps/distgo/cmd/build/build.go

```diff
@@ -245,21 +244,12 @@ func doBuildAction(action buildAction, buildSpec params.ProductBuildSpec, output
 		return errors.Errorf("unrecognized action: %v", action)
 	}
 
-	// execute build args script
-	stdoutBuf := bytes.Buffer{}
-	stderrBuf := bytes.Buffer{}
-	combinedBuf := bytes.Buffer{}
-	stdoutMW := io.MultiWriter(&stdoutBuf, &combinedBuf)
-	stderrMW := io.MultiWriter(&stderrBuf, &combinedBuf)
-	if err := script.WriteAndExecute(buildSpec, buildSpec.Build.BuildArgsScript, stdoutMW, stderrMW, nil); err != nil || stderrBuf.String() != "" {
-		return errors.Wrapf(err, "failed to execute build args script for %v: %v", buildSpec.ProductName, combinedBuf.String())
-	}
-
-	buildArgsString := strings.TrimSpace(stdoutBuf.String())
-	if buildArgsString != "" {
-		buildArgs := strings.Split(buildArgsString, "\n")
-		args = append(args, buildArgs...)
+	// get build args
+	buildArgs, err := script.GetBuildArgs(buildSpec, buildSpec.Build.BuildArgsScript)
+	if err != nil {
+		return errors.Wrapf(err, "")
```

Comment:
If using `Wrapf`, the error should be annotated with extra context. If not, then the `error` can be returned directly without augmentation (this is a big difference between `pkg/errors` and `stacktrace`).